### ISSUE TITLE
ListInstance MajorVersionLimit is never set on SP2013

### ISSUE THREAD

**Issue by lkrn** [SharePoint/PnP-Sites-Core#1943]
#### Category
[ ] Bug
[x] Enhancement

#### Environment
[ ] Office 365 / SharePoint Online
[ ] SharePoint 2016
[x] SharePoint 2013

If SharePoint on-premises, what's exact CU version: 
​15.0.5041.1000

#### Expected or Desired Behavior
When creating a listinstance with the following attributes:

EnableVersioning="true"
MaxVersionLimit="100"

I would expect a list instance to be created with the major version limit set to 100

#### Observed Behavior
Versioning is enabled on the list but the major version limit is not set.

#### Additional Data
Below properties were introduced in Dec 2014 CU for SP2013:
Microsoft.SharePoint.Client.List.MajorVersionLimit
Microsoft.SharePoint.Client.List.MajorWithMinorVersionsLimit
[ref] https://blogs.msdn.microsoft.com/vesku/2014/12/15/latest-api-updates-in-client-side-object-model-dec-2014-cu-for-sp2013/

There's a lot of inconsistency in code for these two.
For example, while MajorVersionLimit is excluded by !SP2013 directive, MajorWithMinorVersionsLimit is used later in the code without any checks (e.g., UpdateList/CreateList methods in ObjectListInstance.cs).

Thus comes the question: are we going to leave in the stone age (I would not be raising this question if these properties were introduced 1 year ago, but it's almost 4 years already) or should we allow setting them on SP2013?
My opinion - if you live on SP2013 pre Dec 2014, you should suffer without using PnP.

If core mainteiners agree to allow them on SP2013, I'll make a PR.
related #1116, affects #1299

**Comment by tmeckel** [comment]
I totally agree with @lkrn to have the `MajorVersionLimit` and `MajorWithMinorVersionsLimit` set for SP2013 because since @VesaJuvonen 's blog post (as of December 15, 2014) the properties are officially supported since Dec 2014 CU for SP2013.  

I can't imagine someone running an SP2013 farm in production with a patch set older than this. So if a provisioning template would crash for such a farm because the properties aren't supported by the CSOM API it should be perfectly okay, because either the user can update the farm to newer CU or modify the template an remove the properties and set them manually via the web interface.
